src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -115,3 +115,16 @@ Variable SparqlQleverVisitor::visitTypesafe(
     SparqlAutomaticParser::VarContext* ctx) {
   return Variable{ctx->getText()};
 }
+
+// ____________________________________________________________________________________
+string SparqlQleverVisitor::visitTypesafe(
+    SparqlAutomaticParser::IriContext* ctx) {
+  string langtag =
+      ctx->PREFIX_LANGTAG() ? ctx->PREFIX_LANGTAG()->getText() : "";
+  if (ctx->iriref()) {
+    return langtag + visitIriref(ctx->iriref()).as<string>();
+  } else {
+    AD_CHECK(ctx->prefixedName())
+    return langtag + visitPrefixedName(ctx->prefixedName()).as<string>();
```

Comment:
Could you (as this is again a small PR where we can perform some cleaning up)
Also move the typesafe parsing one level down (make `iriref` and `prefixedName` typesafe).